app/services/git_tag_push_service.rb

```diff
@@ -2,22 +2,38 @@ class GitTagPushService
   attr_accessor :project, :user, :push_data
 
   def execute(project, user, oldrev, newrev, ref)
-    @project, @user = project, user
-    
-    @push_data = build_push_data(oldrev, newrev, ref)
+    if Gitlab::Git.tag_ref?(ref)
```

Comment:
This is tag push service. This code should be executed only on tag push. What sense to check if it is a tag here?